### PR TITLE
Add support for asdf format as Node.js version file

### PR DESCRIPTION
**Pull request by ganta**
**Description:**
I'm happy to be supported the Node.js version file in #338.
However, we're migrating Node.js version management to asdf.

The version specification file used in asdf has the following format, which is not supported by the current `node-version-file` input:

```
nodejs 14.18.2
```
```
ruby 2.7.4
nodejs 14.18.2
python 3.8.12
```

So, I implemented `node-version-file` to support the format asdf's `.tool-versions` file.

**Related issue:**
#222

**Check list:**
- [x] Mark if documentation changes are required.
- [x] Mark if tests were added or updated to cover the changes.

### CODE REVIEW

**Comment by dmitry-shibanov** [comment]
Hello @ganta. Thank you for your pull request. Could you please revert extra space changes in the yml file and run the `npm run build` command to regenerate `dist` directory.

**Comment by ganta** [comment]
@dmitry-shibanov 
Thank you for your review.

I reverted extra space changes and regenerated dist directory.
Please review again.

Thanks.

**Comment by jokr9** [comment]
joker

File: .github/workflows/versions.yml [hunk 1]
```diff
@@ -84,12 +84,13 @@ jobs:
...
       matrix:
         os: [ubuntu-latest, windows-latest, macos-latest]
+        node-version-file: [.nvmrc, .tool-versions]
     steps:
       - uses: actions/checkout@v2
       - name: Setup node from node version file
         uses: ./
         with:
-          node-version-file: '__tests__/data/.nvmrc'
+          node-version-file: '__tests__/data/${{ matrix.node-version-file }}'
```
Comment (jokr9) [inline .github/workflows/versions.yml:93]: joker hhhhh jhuyrte
iI_SFiS7dtY-arabvid-org mp4_000054610
d1b2ec31-141f-4814-9b28-2882de737f68-640x830

**Comment by dmitry-shibanov** [comment]
Hello @ganta. Sorry for the late response. Could you please sync with the main branch ?

**Comment by ganta** [comment]
I have synced with the main branch and some tests are now failing.
I will fix them.

**Comment by ganta** [comment]
Hi @dmitry-shibanov.

> I have synced with the main branch and some tests are now failing.
> I will fix them.

The test failed because the local environment was M1 Mac.
I have created another Pull Request to resolve that issue. #502 

This Pull Request #373 has passed all checks, could you review it again?

**Review by dmitry-shibanov — APPROVED** [review]

**Review by IvanZosimov — APPROVED** [review]

**Review by marko-zivic-93 — APPROVED** [review]

**Comment by Timmitry** [comment]
Bumping here as this PR would be of great interest for us, too 😇 Anything that is still missing here?

**Review by vsafonkin — APPROVED** [review]

**Comment by andyw8** [comment]
:wave: I'm trying to use this but it seems the node version from `.tool-versions` is not being read, I'm setting it to v19 but it's falling back to v16 (the default). It's working for Ruby but not Node:

https://github.com/andyw8/setup-rails-example-app/blob/main/.tool-versions
https://github.com/andyw8/setup-rails-example-app/actions/runs/3317256754/jobs/5479924455

**Comment by Timmitry** [comment]
@andyw8 I don't see that you are actually setting that file to be used anyhwere. You have to specify

```
- uses: actions/setup-node@v3
  with:
    node-version-file: '.tool-versions'
```

It won't pick that up automatically! See the [documentation](https://github.com/actions/setup-node/blob/main/docs/advanced-usage.md#node-version-file).

**Comment by andyw8** [comment]
Ah, thank you, I had wrongly assumed it would be used automatically since that's how `setup-ruby` behaves.

**Comment by gutwrinch** [comment]
how do i implement Nods.js file into
 uses: actions/setup-node@v3
  with:
    node-version-file: '.tool-versions'

**Comment by gutwrinch** [comment]
is there a help technician that can guide me to through the beginning phases via necessary apps necessary functions for repositories pull requests encode space

**Pull request merged**